src/test/kotlin/com/pubnub/internal/v2/subscription/SubscriptionImplTest.kt

```diff
@@ -0,0 +1,141 @@
+package com.pubnub.internal.v2.subscription
+
+import com.google.gson.JsonNull
+import com.pubnub.api.PNConfiguration
+import com.pubnub.api.PubNub
+import com.pubnub.api.UserId
+import com.pubnub.api.models.consumer.pubsub.BasePubSubResult
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.v2.callbacks.EventListener
+import com.pubnub.api.v2.subscriptions.Subscription
+import com.pubnub.api.v2.subscriptions.SubscriptionOptions
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class SubscriptionImplTest {
+
+    private lateinit var pubnub: PubNub
+    private lateinit var subscription: Subscription
+    private val channelName = "myChannel"
+
+    @BeforeEach
+    fun setUp() {
+        pubnub = PubNub(PNConfiguration(UserId("uuid")))
+        subscription = pubnub.channel(channelName).subscription()
+    }
+
+    @AfterEach
+    fun teardown() {
+        subscription.close()
+        pubnub.destroy()
+    }
+
+    @Test
+    fun plus() {
```

Comment:
I think if possible it would be beneficial to provide more meaningful name for test using should + when or when + should convention e.g.
should add another subscription to a subscription when plus operator method is used
?

Reply:
done